### PR TITLE
[KEYCLOAK-17100] Testsuite Wildfly initialization error on Windows [KEYCLOAK-17392] Java CLASSPATH is wrongly parsed on Windows

### PR DESCRIPTION
**Pull request by iankko**
[KEYCLOAK-17100] Testsuite Wildfly initialization error on Windows
    [KEYCLOAK-17392] Java CLASSPATH is wrongly parsed on Windows
    
    Signed-off-by:  Jan Lieskovsky <jlieskov@redhat.com>
    Co-Authored-By: Peter Zaoral <pzaoral@redhat.com>

Note: Fix the Wildfly initialization error on Windows by removing the support for `-Dwildfly.config.url` parsing in `setJsseSecurityProviderForOutboundSslConnectionsOfElytronClient()` routine. Currently `-Dwildfly.config.url` is / was reused for both the Wildfly initialization, and also for that routine.

But wildfly-client-config performs additional processing of the specified `wildfly.config.url` value (in the `propertyUrlToUri()` routine below):
* https://github.com/wildfly/wildfly-client-config/blob/master/src/main/java/org/wildfly/client/config/ClientConfiguration.java#L225
* https://github.com/wildfly/wildfly-client-config/blob/master/src/main/java/org/wildfly/client/config/ClientConfiguration.java#L227
* https://github.com/wildfly/wildfly-client-config/blob/bce7116e594a8b816bdfea4be67060a56d19a0ad/src/main/java/org/wildfly/client/config/ClientConfiguration.java#L268

Therefore (without the additional processing done in `setJsseSecurityProviderForOutboundSslConnectionsOfElytronClient()`) it wasn't possible to specify a `wildfly.config.url` value (file system path to `wildfly-config.xml`) that would work correctly for both use-cases. And adding that additional processing to `setJsseSecurityProviderForOutboundSslConnectionsOfElytronClient()` isn't an option due to a need of introducing a testsuite dependency on `org.wildfly.client:wildfly-client-config` artifact, which is a no-go option. Another option would be to implement / copy that additional parsing from `wildfly-client-config` code above - but its wrong approach due to code duplication.

Thus fix this by simply dropping the support for `-Dwildfly.config.url` property (which was introduced for ability to override the default `wildfly-config.xml` with custom path) in `AuthServerTestEnricher.java`. With this change the particular location of `wildfly-config.xml` will be identified by scanning the class path like Wildfly does:
  * https://docs.wildfly.org/21/Client_Guide.html#wildfly-config-xml-discovery

Thanks to @Pepo48 for identifying additional [issues](https://issues.redhat.com/browse/KEYCLOAK-17392), suggesting fixes & additional help with the testing the changes on Windows platform.

@pskopek @miquelsi @pdrozd PTAL once got a chance

Thanks, Jan

### CODE REVIEW

**Comment by iankko** [comment]
_Note:_  Would be great to have this merged before the [Upgrade to Wildfly 22 one](https://github.com/keycloak/keycloak/pull/7705).

**Review by miquelsi — DISMISSED** [review]
LGTM. Thanks @iankko and @Pepo48 for the effort!

**Comment by iankko** [comment]
Testing pipeline run:
* Adapter -- http://keycloak.jenkins.server.com/job/universal-test-pipeline-adapters/87/
* Server &nbsp;&nbsp;-- http://keycloak.jenkins.server.com/job/universal-test-pipeline-server/177/

**Comment by iankko** [comment]
> Testing pipeline run:
> 
>     * Adapter -- http://keycloak.jenkins.server.com/job/universal-test-pipeline-adapters/87/
> 
>     * Server   -- http://keycloak.jenkins.server.com/job/universal-test-pipeline-server/177/

Per the pipeline runs, the results are better, but still not completely OK. There are still tests failing with `Failed to locate the wildfly-config.xml to use for the configuration of Elytron client!` message yet (e.g. the `org.keycloak.testsuite.console.BasicConsoleTest` one).

Do Not Merge yet (will remove the draft label, once it's OK). Sorry.

**Comment by iankko** [comment]
Yet different approach:

Keep explicitly pointing to the location of the `wildfly-config.xml` file for those testsuite modules, that don't have it listed as resources (thus it can't be found in the classpath after module build).

But use different property name (`path.to.wildfly.config`), not to collide with Wildfly's / `wildfly-config-clients`'s `wildfly.config.url` property, which is later processed slightly differently (see description of the PR above),

Scanning the classpath for `wildfly-config.xml` is kept as fallback (with KEYCLOAK-17392 fixed).

**Comment by iankko** [comment]
Updated pipeline runs:
* Adapter -- http://keycloak.jenkins.server.com/job/universal-test-pipeline-adapters/90/
* Server &nbsp; -- http://keycloak.jenkins.server.com//job/universal-test-pipeline-server/181/

**Comment by iankko** [comment]
> Updated pipeline runs:
> 
>     * Adapter -- http://keycloak.jenkins.server.com/job/universal-test-pipeline-adapters/90/
> 
>     * Server   -- http://keycloak.jenkins.server.com//job/universal-test-pipeline-server/181/

FWICT from reviewing the test results, the only mention of `wildfly-config.xml` there is just in property definition, e.g. entries like:
```
<property name="path.to.wildfly.config" value="/path/redacted/universal-test-pipeline-adapters/springboot_tests-undertow-default/keycloak/testsuite/integration-arquillian/tests/other/springboot-tests/target/dependency/wildfly-config.xml"/>
```
or the debug message(s) like:
```
07:44:01,411 DEBUG [org.keycloak.testsuite.arquillian.AuthServerTestEnricher] Exploring wildfly-config.xml at the '/path/redacted/universal-test-pipeline-adapters/adapter_tests-wildfly/keycloak/testsuite/integration-arquillian/tests/base/target/dependency/wildfly-config.xml' location...
```
but no errors about failure to locate the `wildfly-config.xml` or initialize the Arquillian testsuite.

So the issue is fixed AFAICT.

**Review by miquelsi — DISMISSED** [review]

File: testsuite/integration-arquillian/tests/pom.xml
Comment (hmlnarik) [inline testsuite/integration-arquillian/tests/pom.xml:662]: Why was this property renamed? Can the name remain as was? `wildfly` is the most determining part of the property and should remain first.

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/AuthServerTestEnricher.java
Comment (hmlnarik) [inline testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/AuthServerTestEnricher.java:605]: No exclamation marks in logs please
```suggestion
                            "Found wildfly-config.xml at '%s' location",
```

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/AuthServerTestEnricher.java
Comment (hmlnarik) [inline testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/AuthServerTestEnricher.java:628]: No need for dots in logs
```suggestion
                "Using wildfly-config.xml from '%s' location",
```

**Review by hmlnarik — CHANGES_REQUESTED** [review]

File: testsuite/integration-arquillian/tests/pom.xml
Comment (Pepo48) [reply testsuite/integration-arquillian/tests/pom.xml:662]: Hi @hmlnarik, Jan explained it in the [comment](https://github.com/keycloak/keycloak/pull/7852#issuecomment-798887126) above. The reason was to avoid a naming collision with wildfly-config-client, which uses a property with the same name but it parses it differently.

I guess we could use something like `<wildfly.config.path>` instead.

Comment (hmlnarik) [reply testsuite/integration-arquillian/tests/pom.xml:662]: Thanks @Pepo48. Would `wildfly-client.config.path` work any better?

Comment (iankko) [reply testsuite/integration-arquillian/tests/pom.xml:662]: Thanks @hmlnarik `wildfly-client.config.path` LGTM

(previously, based on the current content of that `wildfly-config.xml` file wanted to suggest something like `wildfly.elytron.client.config.path`, but given the content of that file might be enhanced in the future yet possibly with other options from [the allowed syntax](https://docs.wildfly.org/23/Client_Guide.html), it's better to use more general purpose name)

For your previous questions:
- We can't use `wildfly.config.url` directly due the reasons in comment \#0 (description) of this PR,
- And `path.to.wildfly.config` was chosen just to avoid a name, analogous to `wildfly.config.url`, which potentially could lead to re-introduction of the same issue (attempt to re-use `wildfly.config.xml`) again.

But `wildfly-client.config.path` LGTM. @Pepo48 What's our take on this?

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/AuthServerTestEnricher.java
Comment (iankko) [reply testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/AuthServerTestEnricher.java]: Removed them (there at the location pointed out, and also at couple of other places in the same routine).

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/AuthServerTestEnricher.java
Comment (iankko) [reply testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/AuthServerTestEnricher.java]: Dropped each instance of it in `testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/AuthServerTestEnricher.java`

File: testsuite/integration-arquillian/tests/pom.xml
Comment (iankko) [reply testsuite/integration-arquillian/tests/pom.xml]: Used `wildfly-client.config.path` with the latest state / change.

Comment (Pepo48) [reply testsuite/integration-arquillian/tests/pom.xml]: @iankko I'm also fine with proposed. Thanks!

**Comment by miquelsi** [comment]
Changes LGTM @iankko. Please make sure most of the checks are passing before merging. Thanks!

**Review by Pepo48 — APPROVED** [review]
LGTM, thanks again @iankko

**Comment by hmlnarik** [comment]
The failures in actions were in download artifact phase, seems like GHA glitch. I've restarted affected pipeline run

**Review by miquelsi — APPROVED** [review]

**Review by hmlnarik — APPROVED** [review]
Thanks. Approving based on @miquelsi and @Pepo48's approvals

**Pull request merged**
